docs/CRD-specification.md

```diff
@@ -0,0 +1,198 @@
+spec:
+  rules:
+  - host: test.apisix.apache.org
+    http:
+      paths:
+      - backend:
+          serviceName: httpserver
+          servicePort: 8080
+        path: /hello*
+        plugins:
+          - name: limit-conn
+            enable: true
+            config:
+              count: 2
+              time_window: 60
+              rejected_code: 503
+              key: remote_addr
+```
+
+|     Field     |  Type    | Description                     |
+|---------------|----------|--------------------------------|
+| rules         | array    | ApisixRoute's request matching rules. |
+| host          | string   | The requested host.        |
+| http          | string   | The routing rule of the request.         |
+| paths         | array    | The routing rule of the request.      |
+| backend       | object   | Backend service information configuration.         |
+| serviceName   | string   | The name of the service.        |
+| servicePort   | int      | The port of the service.        |
+| path          | string   | The URI matched by the route. Supports exact match and prefix match. Example，exact match: `/hello`, prefix match: `/hello/*`.   |
+| plugins       | array    | Custom plugin collection (Plugins defined in the `route` level).    |
+| name          | string   | The name of the plugin.      |
+| enable        | boolean  | Whether to enable the plugin, `true`: means enable, `false`: means disable.      |
+| config        | object   |  Configuration of plugin information.    |
+
+## ApisixService
+
+`ApisixService` corresponds to the `service` object in Apache APISIX.
+
+Structure example:
+
+```yaml
+apiVersion: apisix.apache.org/v1
+kind: ApisixService
+metadata:
+  name: httpserver
+  namespace: cloud  
+spec:
+  upstream: httpserver
+  port: 8080
+  plugins:
+    - name: limit-conn
+      enable: true
+      config:
+        count: 2
+        time_window: 60
+        rejected_code: 503
+        key: remote_addr
+```
+
+|     Field     |  Type    | Description    |
+|---------------|----------|----------------|
+| upstream      | string   | The name of the upstream service.    |
+| port          | int      | The port number of the upstream service.    |
+| plugins       | array   | Custom plugin collection (Plugins defined in the `service` level).  |
+
+## ApisixUpstream
+
+`ApisixUpstream` corresponds to the `upstream` object in Apache APISIX.
+
+Structure example:
+
+```yaml
+apiVersion: apisix.apache.org/v1
+kind: ApisixUpstream
+metadata:
+  name: httpserver
+  namespace: cloud
+spec:
+  ports:
+    - port: 8080
+      loadbalancer: roundrobin
+```
+
+|     Field     |  Type    | Description    |
+|---------------|----------|----------------|
+| ports         | array    | Custom upstream collection.   |
+| port          | int      | Upstream service port number.    |
+| loadbalancer  | string/object   | Select the algorithm of the upstream service, support `roundrobin` and `chash`.  |
+
+## ApisixPlugin
```

Comment:
We don't have this CRD.

Reply:
I saw that this CRD design was included, so I added it.
[https://github.com/apache/apisix-ingress-controller/issues/3 .](https://github.com/apache/apisix-ingress-controller/issues/3).

Do we need to delete this CRD?